server/src/repositories/Equipment/ReservationRepository.ts

```diff
@@ -54,32 +64,129 @@ export class ReservationRepository implements IReservationRepository {
         await this.queryBuilder("Reservations")
         .where("id", id)
         .update({
-            userId: reservation.userId,
-            supervisorId: reservation.supervisorId,
-            machineId: reservation.machineId,
-            startTime: reservation.startTime,
-            endTime: reservation.endTime
+          creator: reservation.creator,
+          labbie: reservation.labbie,
+          maker: reservation.maker,
+          equipment: reservation.equipment,
+          startTime: reservation.startTime,
+          endTime: reservation.endTime,
+          startingMakerComment: reservation.startingMakerComment
         });
         return this.getReservationById(id);
     }
 
-    public async addReservation(reservation: ReservationInput): Promise<Reservation | null> {
-      const newId = (
-        await this.queryBuilder("Reservations").insert(
-          {
-            userId: reservation.userId,
-            supervisorId: reservation.supervisorId,
-            machineId: reservation.machineId,
-            startTime: reservation.startTime,
-            endTime: reservation.endTime
-          },
-          "id"
-        )
-      )[0];
-      return await this.getReservationById(newId);
+    public async createReservation(reservation: ReservationInput): Promise<Reservation | null> {
+      // get all modules needed for the equipment
+      const modules = (
+        await this.queryBuilder("ModulesForEquipment")
+        .select("trainingModuleId")
+        .where("equipmentId", reservation.equipment)
+      );
+      let passed = true;
+      // get last submission from maker for every module
+      modules.forEach(async (module: any) => {
+        const eligibility = await this.queryBuilder("ModuleSubmissions")
+          .select("passed")
+          .where("moduleID", module.id)
+          .where("makerID", reservation.maker)
+          .orderBy("submissionDate", "desc")
+          .limit(1)
+          .first();
+        if (!eligibility) {
+          passed = false;
+        }
+      });
+      // in progress
+      // const startTime = reservation.startTime;
+      // const endTime
+      // const free = await this.queryBuilder("Reservations")
+      //   .select("*")
+      //   .whereBetween("startTime", [reservation.startTime, reservation.endTime])
+      //   .whereBetween("endTime", [reservation.startTime, reservation.endTime])
+      // if maker has passed all modules 
+      if (passed) {
```

Comment:
I think in general we want to limit the repositories to simple database operations, and put the business logic in the resolvers.

CC: @pjw9963 for confirmation